cf_il/train.py

```diff
@@ -0,0 +1,102 @@
+# Based on training.py file
+
+from argparse import Namespace
+from typing import List
+
+import numpy as np
+
+from cf_il.model import CFIL
+from datasets.utils.continual_dataset import ContinualDataset
+from training import evaluate
+from utils.loggers import print_mean_accuracy
+from utils.status import progress_bar, create_stash
+from utils.tb_logger import TensorboardLogger
+
+
+def train(
+    model: CFIL,
+    dataset: ContinualDataset,
+    args: Namespace,
+) -> None:
+    """
+    The training process, including evaluations and loggers.
+    :param model: the module to be trained
+    :param dataset: the continual dataset at hand
+    :param args: the arguments of the current execution
+    """
+    assert dataset.N_TASKS is not None
+    assert dataset.N_CLASSES_PER_TASK is not None
+
+    # TODO: change to some resonable restriction
+    # Currently model is recovering `num_images_per_class` images for each class
+    # where it is set as args.minibatch_size below.
+    assert args.buffer_size >= args.minibatch_size * dataset.N_CLASSES_PER_TASK
+
+    model.net.to(model.device)
+
+    num_tasks: int = dataset.N_TASKS
+    num_classes_per_task: int = dataset.N_CLASSES_PER_TASK
+    results: List[List[float]] = []
+    results_mask_classes: List[List[float]] = []
+
+    model_stash = create_stash(model, args, dataset)
+
+    if args.tensorboard:
+        tb_logger = TensorboardLogger(args, dataset.SETTING, model_stash)
+        model_stash['tensorboard_name'] = tb_logger.get_name()
+
+    for task_idx in range(num_tasks):
+        model.net.train()
```

Comment:
to chyba można przenieść poza pętlę, nie ma sensu tego wywoływać za każdym razem

Reply:
chodziło mi aby linię 49 przenieść do 47, bo tak za każdym razem to wołamy (choć do póki nie ma nigdzie ustawiania na val to można to zrobić, ale jak już będzie to musi tak zostać, także w sumie lepiej jak zostanie xd)